lib/project_types/script/layers/application/push_script.rb

```diff
@@ -14,11 +14,19 @@ def call(ctx:, language:, extension_point_type:, script_name:, source_file:, api
             task_runner = Infrastructure::TaskRunner.for(ctx, language, script_name, source_file)
             ProjectDependencies.install(ctx: ctx, task_runner: task_runner)
             BuildScript.call(ctx: ctx, task_runner: task_runner, script: script)
+            check_if_safe_to_push(ctx, task_runner)
             push_script(ctx, task_runner, script, api_key, force)
           end
 
           private
 
+          def check_if_safe_to_push(ctx, task_runner)
+            UI::PrintingSpinner.spin(ctx, ctx.message('script.application.checking')) do |p_ctx, spinner|
+              task_runner.check_if_safe_to_push!
+              spinner.update_title(p_ctx.message('script.application.checked'))
+            end
+          end
+
```

Comment:
A few comments here:
1. Should we do this check when we check dependencies?  Currently, we only check node version, and it feels like checking the versions of other dependencies is a similar task.  Also, that will allow this `push` command to fail earlier.  Why build, if essentially, you cannot push.  Imagine the developer is fixing all the syntax errors, writing unit tests, and got pass the build phase, only realize that he has been coding against an out-of-date API.  That could be frustrating?  
2. Are you considering adding this check to `enable` command as well?  Imagine if a developer has already pushed the Script to prod a day ago when it wasn't out of date, and today he comes back to just enable it.  That `enable` command will succeed, but he will not know that the API is now out of date.

Reply:
For (1), that's a good point. I was thinking about more as this probably shouldn't fail the build since it's nothing wrong with the build but I agree that we can check in the CLI before even attempting to build. I can move it into the check dependencies logic.

For (2), I was not. I'm not understanding why that would be necessary. When a script is enabled shouldn't have any impact on data serializing or deserializing correctly. When it's pushed currently would unfortunately. The API being out of date is not exactly the issue, it's more there's some stuff that happens server-side specifically at push time that can break serialization if the EP package isn't up-to-date. I can go over it in more detail over Slack or a call, mostly just don't want to go too deep into the server implementation details while commenting in a public repository.